### PR TITLE
Jv-fruit-shop-test solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

**Comment by mkazarian-1** [comment]
I want to ask you to check my jv-fruit-shop part without test because during the testing I have found some mistakes.

And also I want to apologize for not separating the main code, main code fix and tests into different commits. I realized it too late.

File: src/test/java/core/basesyntax/resources/test_report.csv
Comment (Rommelua) [inline src/test/java/core/basesyntax/resources/test_report.csv:1]: image

File: src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class FileReaderCvsImplTest {
+    private static final String TEMP_TEST_FILE_PATH =
+            "src/test/java/core/basesyntax/resources/test_activities.csv";
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java:18]: image

File: src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    @BeforeAll
+    public static void initAll() {
+        try {
+            Files.writeString(Paths.get(TEMP_TEST_FILE_PATH), "");
+        } catch (IOException e) {
+            throw new RuntimeException("Error cleaning data in file: " + TEMP_TEST_FILE_PATH);
+        }
+    }
+
+    private void writeDataInFile(String text) {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java:30]: Private ones (methods, classes) should be after public ones.

```suggestion
    private void writeDataToFile(String text) {
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/dao/FruitTransactionDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+        Map<String, Integer> nameQuantityMap = new HashMap<>();
+        nameQuantityMap.put("apple", 10);
+        nameQuantityMap.put("something", 10);
+
+        assertEquals(Storage.fruits, nameQuantityMap);
+    }
+
+    @Test
+    public void update_nullField_ok() {
```
Comment (MAMentorFX) [inline src/test/java/core/basesyntax/dao/FruitTransactionDaoImplTest.java:49]:
```suggestion
    public void update_nullField_throwsException() {
```
I don't think that is `ok`, please make sure all test methods are named in correct way

File: src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        try {
+            Files.writeString(Paths.get(TEMP_TEST_FILE_PATH), "");
+        } catch (IOException e) {
+            throw new RuntimeException("Error cleaning data in file: " + TEMP_TEST_FILE_PATH);
+        }
+    }
+
+    @Test
+    public void read_generalData_ok() {
+        List<String> expect = new ArrayList<>();
```
Comment (MAMentorFX) [inline src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java:32]:
```suggestion
        List<String> expected = new ArrayList<>();
```
rename all expect to expected

File: src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+        writeDataInFile(convertListToString(expect));
+
+        List<String> actual = fileReader.read(TEMP_TEST_FILE_PATH);
+
+        assertEquals(expect, actual);
+    }
+
+    @Test
+    public void read_wrongPath_notOk() {
```
Comment (MAMentorFX) [inline src/test/java/core/basesyntax/service/impl/FileReaderCvsImplTest.java:74]:
```suggestion
    public void read_wrongPath_thowsException() {
```
be more specific

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by kerrrusha — APPROVED** [review]

**Pull request closed without merging**